### PR TITLE
[W5.11][W13-2] Arjo Chakravarty

### PR DESCRIPTION
**Pull request by arjo129**
Added edit_name command

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -151,6 +151,14 @@ Views all details of the 2nd person in the address book.
 `viewall 1` +
 Views all details of the 1st person in the results of the `find` command.
 
+== Edit name : `edit_name`
```
Comment (okkhoy) [inline docs/UserGuide.adoc:154]: 👍 for updating the user guide

File: src/seedu/addressbook/commands/EditNameCommand.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/EditNameCommand.java:5]: Don't import everything using `*`. Import the necessary classes explicitly.

File: src/seedu/addressbook/commands/EditNameCommand.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            + "Example: " + COMMAND_WORD
+            + " John Doe n/New Name";
+    private static final String MESSAGE_DUPLICATE = "Cannot update address book to same name";
+    public static final String MESSAGE_SUCCESS = "Person updated: %1$s";
+    private String oldPersonName, newPersonName;
+    public EditNameCommand(String name, String newName){
+        this.oldPersonName = name;
+        this.newPersonName = newName;
+    }
+    @Override
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/EditNameCommand.java:20]: Could use some vertical spacing for readability

File: src/seedu/addressbook/commands/EditNameCommand.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        try {
+            Person newEntry = new Person(
+                    new Name(this.newPersonName),
+                    toBeRemoved.getPhone(),
+                    toBeRemoved.getEmail(),
+                    toBeRemoved.getAddress(),
+                    toBeRemoved.getTags()
+            );
+            addressBook.addPerson(newEntry);
+            addressBook.removePerson(toBeRemoved);
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/EditNameCommand.java:41]: Think how you can do in place edits

**Review by okkhoy — APPROVED** [review]
Good enhancement!

**Comment by okkhoy** [comment]
@arjo129 
Some comments added. Please take note and close the pull request.

**Comment by arjo129** [comment]
Thank you

**Pull request closed without merging**